repose-aggregator/src/docs/asciidoc/recipes/functional-test-framework.adoc

```diff
@@ -0,0 +1,167 @@
+= Functional Test Framework
+
+The Repose functional test framework is a testing library provided by the Repose team meant to simplify the process of creating functional tests for features in Repose or its extension components.
+
+This framework is based on the http://spockframework.org/[Spock framework].
+While there are features of this framework that are can be used without Spock, it is expected that tests utilizing this framework will be Spock tests.
+
+== Download
+
+The Repose functional test framework is published to https://maven.research.rackspacecloud.com/[an instance of Nexus hosted by Rackspace].
+
+It is highly recommended that a build system be used to manage this library.
+The following snippets provide the necessary additions to download this framework with some popular build systems.
+
+=== Maven
+
+[source,xml]
+.pom.xml
+----
+<repositories>
+    <repository>
+        <id>public.maven.research.rackspace.com</id>
+        <name>Rackspace Research</name>
+        <url>https://maven.research.rackspacecloud.com/content/groups/aggregate</url>
+    </repository>
+</repositories>
+
+<dependencies>
+    <dependency>
+      <groupId>org.openrepose</groupId>
+      <artifactId>functional-test-framework</artifactId>
+      <version>${version}</version>
+    </dependency>
+</dependencies>
+----
+
+=== Gradle
+
+[source,groovy]
+.build.gradle
+----
+repositories {
+    maven {
+        url "https://maven.research.rackspacecloud.com/content/groups/aggregate/"
+    }
+}
+
+dependencies {
+    compile "org.openrepose:functional-test-framework:${version}"
+}
+----
+
+== Setup
+
+There are two requirements that must be satisfied for this framework.
+
+The first requirement is that a `repose-test.properties` file is on the classpath when tests are run.
+The `repose-test.properties` file tells that framework what version of Repose, directories, host, and ports to use.
+It should be in the https://en.wikipedia.org/wiki/.properties#Format[Java properties file format] like the following example:
+
+[NOTE]
+====
+Not all properties are strictly required.
+In fact, most properties have reasonable defaults.
+However, any property which takes an absolute path should be given a value.
+====
+
+----
+# Required Properties
+test.root.directory=/default/test/root # <1>
+
+repose.home=/default/repose/home # <2>
+repose.config.templates=/default/config/templates # <3>
+
+repose.jar=/default/valve/location # <4>
+repose.lint.jar=/default/lint/location # <5>
+repose.root.war=/default/root/war/location # <6>
+glassfish.jar=/default/glassfish/location # <7>
+tomcat.jar=/default/tomcat/location # <8>
+mocks.war=/default/mocks/war/location # <9>
```

Comment:
still need to call out that you only need to provide those that you are using. If you aren't doing anything with tomcat or glassfish it isn't required.